content/docs/cookbooks/tls-auth.md

```diff
@@ -0,0 +1,46 @@
+---
+title: Securing Prometheus endpoints using TLS authentication
+sort_rank: 1
+---
+
+# Securing Prometheus endpoints using TLS auth
+
+Prometheus does *not* directly support authentication via [Transport Layer Security](https://en.wikipedia.org/wiki/Transport_Layer_Security), and this is so [by design](/docs/introduction/faq/#tls). In fact, Prometheus provides no server-side authentication, authorization, or encryption.
+
+Nonetheless, we do strongly recommended that you use TLS or some other form of authentication when using Prometheus by applying it via reverse proxy. You can use any proxy you like with Prometheus. In this cookbook, however, we'll cover only [nginx](#nginx).
+
+NOTE: For more information on the reasoning behind these and other security-related decisions in Prometheus, see the [Security Model](/docs/operating/security) documentation.
+
+## nginx example
+
+CAUTION: The nginx configuration below is just an example. We strongly recommend consulting the official [nginx documentation](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_certificate) when creating a customized configuration, deciding on TLS protocols and ciphers to use, and so on.
+
+Let's say that you're running Prometheus behind an [nginx](https://www.nginx.com/) server running at https://www.my-nginx-proxy.org and you want to access Prometheus via the `/prometheus` endpoint. Here's an example cURL command that would query the Prometheus server's `/metrics` endpoint:
+
+```bash
+$ curl https://my-nginx-proxy.org/prometheus/metrics
+```
+
+Let's also say that you've generated the following using [OpenSSL](https://www.openssl.org/) or an analogous tool:
+
+* an SSL certificate at `/root/certs/my-nginx-proxy.com/my-nginx-proxy.com.crt`
+* an SSL key at `/root/certs/my-nginx-proxy.com/my-nginx-proxy.com.key`
+
+Below is an example nginx configuration that would handle TLS authentication at the nginx level and then proxy all connections on port 443 to a Prometheus server running on the same host:
+
+```nginx
+http {
+  server {
+    listen              443 ssl;
+    server_name         my-nginx-proxy.com www.my-nginx-proxy.com;
+    ssl_certificate     /root/certs/my-nginx-proxy.com/my-nginx-proxy.com.crt;
+    ssl_certificate_key /root/certs/my-nginx-proxy.com/my-nginx-proxy.com.key;
+    ssl_protocols       TLSv1 TLSv1.1 TLSv1.2;
+    ssl_ciphers         AES128-SHA:AES256-SHA:RC4-SHA:DES-CBC3-SHA:RC4-MD5;
+
+    location /prometheus-metrics {
```

Comment:
No mention of --web.external-url to Prometheus? This setup won't work without it

Reply:
The question mark here is unnecessary. I’ve been learning about Prometheus for a little over a week and this is not obvious.

Reply:
I'm sorry if I came across badly, I know it's not obvious which is why I brought it up.

Reply:
No worries 👍

Reply:
"Make sure to mention the [`--web.external-url`](https://github.com/prometheus/prometheus/blob/master/cmd/prometheus/main.go#L125) flag for Prometheus here. Without that flag, the proxy won't be able to reach the running Prometheus instance."

Reply:
I'd give a full example, here it'd be https://localhost/promethus/

Reply:
With this particular setup you also need to set ---web.route-prefix=/, or to avoid that you can proxy to http://localhost:900/prometheus

Reply:
Yeah, I’m thinking maybe I’ll make the practical example all localhost and then advise to replace localhost with the “real” address.